File: clientapp/src/validators/EmailValidator.ts
import regExpValidator from "./RegExpValidator";

export default function emailValidator(email: string): boolean {
  return regExpValidator(
    /^[A-Za-z0-9._%+-]{1,20}@[A-Za-z0-9.-]{1,20}\.[A-Z|a-z]{2,4}$/,
    email
  );
}

File: serverApp/Validations/EmailValidator.cs
using FluentValidation;

public class EmailValidator : AbstractValidator<string>
{
    public EmailValidator()
    {
        RuleFor(x => x).NotEmpty().EmailAddress().WithMessage("Email not valid");
    }
}

public static class EmailStaticValidator
{
    public static bool IsValid(string email)
    {
        return new EmailValidator().Validate(email).IsValid;
    }
}